### PR TITLE
PatchEngine: check instruction at link addr is non-zero

### PR DESCRIPTION
**Pull request by ligfx**
Regression introduced in #4738. Fixes https://bugs.dolphin-emu.org/issues/10110.

### CODE REVIEW

File: Source/Core/Core/PatchEngine.cpp [hunk 1]
```diff
@@ -223,7 +223,8 @@ static bool IsStackSane()
     return false;
 
   // Check the link register makes sense (that it points to a valid IBAT address)
-  return PowerPC::HostIsInstructionRAMAddress(PowerPC::HostRead_U32(next_SP + 4));
+  u32 address = PowerPC::HostRead_U32(next_SP + 4);
+  return PowerPC::HostIsInstructionRAMAddress(address) && PowerPC::HostRead_U32(address);
```
Comment (lioncash) [inline Source/Core/Core/PatchEngine.cpp]: Please use explicit boolean operators with arithmetic values (in this case `!= 0`).

File: Source/Core/Core/PatchEngine.cpp [hunk 2]
```diff
@@ -223,7 +223,8 @@ static bool IsStackSane()
     return false;
 
   // Check the link register makes sense (that it points to a valid IBAT address)
-  return PowerPC::HostIsInstructionRAMAddress(PowerPC::HostRead_U32(next_SP + 4));
+  u32 address = PowerPC::HostRead_U32(next_SP + 4);
```
Comment (lioncash) [inline Source/Core/Core/PatchEngine.cpp]: This can be const

Comment (ligfx) [inline Source/Core/Core/PatchEngine.cpp]: Nice catch

File: Source/Core/Core/PatchEngine.cpp (hunk 1 above)
Comment (ligfx) [inline Source/Core/Core/PatchEngine.cpp]: Done.

**Comment by Parlane** [comment]
@dolphin-emu-bot rebuild

**Pull request merged**